packages/back-end/src/routers/organizations/organizations.controller.ts

```diff
@@ -1070,6 +1074,121 @@ export async function postImportConfig(
   });
 }
 
+export async function getOrphanedUsers(req: AuthRequest, res: Response) {
+  req.checkPermissions("organizationSettings");
+
+  if (IS_CLOUD) {
+    throw new Error("Unable to get orphaned users on GrowthBook Cloud");
+  }
+
+  const allUsers = await getAllUsers();
+  const allOrgs = await findAllOrganizations();
+
+  const membersInOrgs = new Set<string>();
+  allOrgs.forEach((org) => {
+    org.members.forEach((m) => {
+      membersInOrgs.add(m.id);
+    });
+  });
+
+  const orphanedUsers = allUsers
+    .filter((u) => !membersInOrgs.has(u.id))
+    .map(({ id, name, email }) => ({
+      id,
+      name,
+      email,
+    }));
+
+  return res.status(200).json({
+    status: 200,
+    orphanedUsers,
+  });
+}
+
+export async function addOrphanedUser(
+  req: AuthRequest<MemberRoleWithProjects, { id: string }>,
+  res: Response
+) {
+  req.checkPermissions("organizationSettings");
+
+  const { org } = getOrgFromReq(req);
+
+  const { id } = req.params;
+  const {
+    role,
+    environments,
+    limitAccessByEnvironment,
+    projectRoles,
+  } = req.body;
+
+  // Make sure user exists
+  const user = await findUserById(id);
+  if (!user) {
+    return res.status(400).json({
+      status: 400,
+      message: "Cannot find user with that id",
+    });
+  }
+
+  // Make sure user is actually orphaned
+  const orgs = await findOrganizationsByMemberId(id);
+  if (orgs.length) {
+    return res.status(400).json({
+      status: 400,
+      message: "Cannot add users who are already part of an organization",
+    });
+  }
+
+  await addMemberToOrg({
+    organization: org,
+    userId: id,
+    role,
+    environments,
+    limitAccessByEnvironment,
+    projectRoles,
+  });
```

Comment:
It looks like this allows any authenticated user with `organizationSettings` permission (for any organization) to add any orphaned user to their org. I'm not sure if there's an issue with doing that, but should this functionality maybe also be restricted to `IS_CLOUD = false` like the other methods?

Reply:
Yes, good catch